[Bug]: Parent market forgets successor market during checkpoint restart

### Problem encountered

If you create a market and a successor market, then perform a checkpoint restart, the parent no longer lists the child as a successor post restart (although the child still lists the parent)

### Observed behaviour

        assert marketsTrading.get_market(child1.id)["parentMarketId"] == parent1.id
>       assert marketsTrading.get_market(parent1.id)["successorMarketId"] == child1.id
E       KeyError: 'successorMarketId'

tests/governance/test_markets.py:875: KeyError

### Expected behaviour

Parent and child remember each other.

### Steps to reproduce

```bash
1. Check out https://github.com/vegaprotocol/system-tests/tree/jake_successor_market_cont
2. pytest -k test_settled_markets_saved_in_checkpoint_for_time_window
```


### Software version

fef2b0d70bccde2bc695787f2fe27330472f1f2b

### Failing test

_No response_

### Jenkins run

https://jenkins.ops.vega.xyz/job/common/job/system-tests-wrapper/85151/

### Configuration used

_No response_

### Relevant log output

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -59,8 +59,7 @@
 - [8477](https://github.com/vegaprotocol/vega/issues/8477) - Do not allow opening auction duration of 0
 - [8891](https://github.com/vegaprotocol/vega/issues/8891) - Emit market update event when resuming via governance
 - [8874](https://github.com/vegaprotocol/vega/issues/8874) - Database migration can fail when rolling back and migrating up again.
--
-
+- [8855](https://github.com/vegaprotocol/vega/issues/8855) - Preserve reference to parent market when restoring checkpoint data
 
 ## 0.72.1
 

--- a/core/execution/engine.go
+++ b/core/execution/engine.go
@@ -364,7 +364,12 @@ func (e *Engine) succeedOrRestore(ctx context.Context, successor, parent string,
 	// restoring a market, but no state of the market nor parent market exists. Treat market as parent.
 	if restore && !sok && !ok {
 		// restoring a market, but the market state and parent market both are missing
+		// this market, upon leaving opening auction, cannot possibly succeed a market that no longer exists
+		// now we should reset
 		mkt.ResetParentIDAndInsurancePoolFraction()
+		// remove from maps
+		delete(e.successors, parent)
+		delete(e.isSuccessor, successor)
 		return nil
 	}
 	// if parent market is active, mark as succeeded
@@ -424,9 +429,11 @@ func (e *Engine) RestoreMarket(ctx context.Context, marketConfig *types.Market) 
 		if len(marketConfig.ParentMarketID) == 0 {
 			return nil
 		}
-		// check to see if the parent market can be found, remove if the parent market is gone
+		// check to see if the parent market can be found, remove from the successor maps if the parent is gone
+		// the market itself should still hold the reference because state was restored
 		if _, ok := e.futureMarkets[marketConfig.ParentMarketID]; !ok {
-			e.futureMarkets[marketConfig.ID].ResetParentIDAndInsurancePoolFraction()
+			delete(e.successors, marketConfig.ParentMarketID)
+			delete(e.isSuccessor, marketConfig.ID)
 		}
 		return nil
 	}

